src/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect } from 'react';
+
+// from https://usehooks.com/useOnClickOutside/
+export function useOnClickOutside(
+  ref: React.RefObject<HTMLElement>,
+  handler: Function
+) {
+  useEffect(
+    () => {
+      //@ts-ignore
+      const listener = event => {
+        // Do nothing if clicking ref's element or descendent elements
+        if (!ref.current || ref.current.contains(event.target)) {
+          return;
+        }
+        handler(event);
+      };
+
+      document.addEventListener('mousedown', listener);
+
+      document.addEventListener('touchstart', listener);
+
+      return () => {
+        document.removeEventListener('mousedown', listener);
+
+        document.removeEventListener('touchstart', listener);
+      };
+    },
+
+    // Add ref and handler to effect dependencies
+
+    // It's worth noting that because passed in handler is a new ...
```

Comment:
@jb-san I'm a bit confused about the leading and trailing dots, any special reason for it?

Reply:
nope, thats just how it was written on useHooks.com